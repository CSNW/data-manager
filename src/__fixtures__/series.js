export function single() {
  return [
    {
      values: [
        { a: 0, b: 10 },
        { a: 1, b: 8 },
        { a: 2, b: 6 },
        { a: 3, b: 4 },
        { a: 4, b: 2 },
        { a: 5, b: 0 }
      ]
    }
  ];
}

export function multi() {
  return [single()[0], single()[0], single()[0]];
}
