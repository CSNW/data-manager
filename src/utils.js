function numbersAscending(a, b) {
  return a - b;
}
function numbersDescending(a, b) {
  return b - a;
}

export const compare = {
  numbersAscending,
  numbersAsc: numbersAscending,
  numbersDescending,
  numbersDesc: numbersDescending
};

export function mapValues(data, iterator) {
  return data.map(series => {
    const mapped = shallowCloneObj(series);
    mapped.values = iterator(series.values);

    return mapped;
  });
}

export function shallowCloneObj(obj) {
  const cloned = {};
  for (const key in obj) {
    cloned[key] = obj[key];
  }
  return cloned;
}
