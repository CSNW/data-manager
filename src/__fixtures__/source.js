import Source from '../source';
import { single } from './series';

export default function source() {
  const override = new Source();
  override.fetch = jest.fn(() => Promise.resolve(single()[0].values));

  return override;
}
