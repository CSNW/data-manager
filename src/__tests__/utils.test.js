import { compare, mapValues, shallowCloneObj } from '../utils';
import * as series from '../../tests/__fixtures__/series';

describe('compare', () => {
  test('should sort numbers ascending', () => {
    expect([2, 4, 1, 3, 5].sort(compare.numbersAscending)).toMatchSnapshot();
    expect([2, 4, 1, 3, 5].sort(compare.numbersAsc)).toMatchSnapshot();
  });

  test('should sort numbers descending', () => {
    expect([2, 4, 1, 3, 5].sort(compare.numbersDescending)).toMatchSnapshot();
    expect([2, 4, 1, 3, 5].sort(compare.numbersDesc)).toMatchSnapshot();
  });
});

describe('mapValues', () => {
  test('should map series values', () => {
    expect(
      mapValues(series.multi(), values => values.filter(row => row.a > 2))
    ).toMatchSnapshot();
  });

  test('should clone series', () => {
    const original = series.multi();
    const result = mapValues(original, values =>
      values.filter(row => row.a <= 2)
    );

    expect(original[0]).not.toBe(result[0]);
  });
});

describe('shallowCloneObj', () => {
  test('should clone object fields', () => {
    const original = { a: 1, b: { c: 3 } };
    const cloned = shallowCloneObj(original);

    expect(cloned).not.toBe(original);
    expect(cloned).toEqual(original);
    expect(cloned.b).toBe(original.b);
  });
});
