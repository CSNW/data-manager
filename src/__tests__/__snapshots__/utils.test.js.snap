// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare should sort numbers ascending 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;

exports[`compare should sort numbers ascending 2`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;

exports[`compare should sort numbers descending 1`] = `
Array [
  5,
  4,
  3,
  2,
  1,
]
`;

exports[`compare should sort numbers descending 2`] = `
Array [
  5,
  4,
  3,
  2,
  1,
]
`;

exports[`mapValues should map series values 1`] = `
Array [
  Object {
    "values": Array [
      Object {
        "a": 3,
        "b": 4,
      },
      Object {
        "a": 4,
        "b": 2,
      },
      Object {
        "a": 5,
        "b": 0,
      },
    ],
  },
  Object {
    "values": Array [
      Object {
        "a": 3,
        "b": 4,
      },
      Object {
        "a": 4,
        "b": 2,
      },
      Object {
        "a": 5,
        "b": 0,
      },
    ],
  },
  Object {
    "values": Array [
      Object {
        "a": 3,
        "b": 4,
      },
      Object {
        "a": 4,
        "b": 2,
      },
      Object {
        "a": 5,
        "b": 0,
      },
    ],
  },
]
`;
